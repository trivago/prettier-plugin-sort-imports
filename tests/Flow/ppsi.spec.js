run_spec(__dirname, ['flow'], {
    importOrder: ['^@core/(.*)$', '^@server/(.*)', '^@ui/(.*)$', '^[./]'],
    importOrderSeparation: true,
});
