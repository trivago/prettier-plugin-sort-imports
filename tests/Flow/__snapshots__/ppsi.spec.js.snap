// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports-with-comments.js - flow-verify: imports-with-comments.js 1`] = `
/**
 * @flow
 */

// I am top level comment in this file.
import { type Something } from './__generated__/';
import threeLevelRelativePath from "../../../threeLevelRelativePath";
import sameLevelRelativePath from "./sameLevelRelativePath";
import thirdParty from "third-party";
import oneLevelRelativePath from "../oneLevelRelativePath";
import otherthing from "@core/otherthing";
import abc from "@core/abc";
import twoLevelRelativePath from "../../twoLevelRelativePath";
import component from "@ui/hello";
import fourLevelRelativePath from "../../../../fourLevelRelativePath";
import something from "@server/something";
import xyz from "@ui/xyz";

const handleChange = (value: ?string) => {}

export type AliasFoo3  = {
  givesANum(): number
};
export function givesAFoo3Obj(): AliasFoo3 {
  return {
    givesANum(): number { return 42; }
  };
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/**
 * @flow
 */
// I am top level comment in this file.
import thirdParty from "third-party";

import abc from "@core/abc";
import otherthing from "@core/otherthing";

import something from "@server/something";

import component from "@ui/hello";
import xyz from "@ui/xyz";

import fourLevelRelativePath from "../../../../fourLevelRelativePath";
import threeLevelRelativePath from "../../../threeLevelRelativePath";
import twoLevelRelativePath from "../../twoLevelRelativePath";
import oneLevelRelativePath from "../oneLevelRelativePath";
import { type Something } from "./__generated__/";
import sameLevelRelativePath from "./sameLevelRelativePath";

const handleChange = (value: ?string) => {};

export type AliasFoo3 = {
    givesANum(): number,
};
export function givesAFoo3Obj(): AliasFoo3 {
    return {
        givesANum(): number {
            return 42;
        },
    };
}

`;
